### ISSUE TITLE
`row.pop(None)` type error when working with rows from `DictReader`

### ISSUE THREAD

**Issue by simonw** [python/typeshed#8075]
`csv.DictReader` returns rows that include a key of `None` if you feed it a CSV file where some of the rows have more items than exist in the header:

```python
import csv, io
rows = csv.DictReader(io.StringIO("id,name\n1,Cleo,extra"))
list(rows)
```
Outputs:
```
[{'id': '1', 'name': 'Cleo', None: ['extra']}]
```
I wrote some code that used `row.pop(None)` to remove that last key - but I got a type error from `mypy`.

Here's a simple `demo.py` module showing the error:

```python
import csv, io

rows = csv.DictReader(io.StringIO("id,name\n1,Cleo,extra"))
for row in rows:
    if None in row:
        row.pop(None)
    print(row)
```
When I run `mypy demo.py` I get this:
```
demo.py:6: error: No overload variant of "pop" of "MutableMapping" matches argument type "None"
demo.py:6: note: Possible overload variants:
demo.py:6: note:     def pop(self, key: str) -> str
demo.py:6: note:     def [_T] pop(self, key: str, default: Union[str, _T] = ...) -> Union[str, _T]
Found 1 error in 1 file (checked 1 source file)
```

**Comment by JelleZijlstra** [comment]
See also #7953.

**Comment by Akuli** [comment]
This was fixed in #8061, but the fix hasn't been released yet.

**Comment by AlexWaygood** [comment]
> See also #7953.

Yes: I think this false-positive is already fixed on typeshed master following #8061 (use the `--custom-typeshed-dir` command-line argument to try using mypy with typeshed's master branch). Mypy hasn't done a typeshed sync in a few weeks, but it should be fixed on mypy master as soon as they do so :)
